pkg/admission/customresourcevalidation/image/validate_image.go

```diff
@@ -0,0 +1,100 @@
+package image
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/origin/pkg/admission/customresourcevalidation"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register("config.openshift.io/ValidateImage", func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				configv1.Resource("images"): true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				configv1.GroupVersion.WithKind("Image"): imageV1{},
+			})
+	})
+}
+
+func toImageV1(uncastObj runtime.Object) (*configv1.Image, field.ErrorList) {
```

Comment:
do we need this boilerplate in every validation? Can't we check in a generic way in the framework code, if necessary by passing the target kind there.

Reply:
> do we need this boilerplate in every validation? Can't we check in a generic way in the framework code, if necessary by passing the target kind there.

Probably via reflection.  I'm open to future refinement.

Reply:
> Probably via reflection. I'm open to future refinement.

you have to use reflection for the actual method calls